File: app/src/main/java/com/example/rahulmishra/minesweeper/MyButton.java
package com.example.rahulmishra.minesweeper;

import android.content.Context;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.AppCompatButton;
import android.widget.Button;

/**
 * Created by Rahul Mishra on 17-06-2017.
 */

public class MyButton extends AppCompatButton {
    int value = 0;
    boolean isFlagged = false;
    boolean isVisible = false;
    int validFlagCounter = 0;
    int row;
    int col;
    Context c ;

    public MyButton(Context context) {

        super(context);
        c = context ;
    }

    public void applyChanges(int state)  {
        switch (state)  {
            case ButtonConstants.DEFAULT :
                super.setBackgroundDrawable(ContextCompat.getDrawable(getContext(), R.drawable.button));
                break ;
            case ButtonConstants.FALSEFLAG :
            case ButtonConstants.ZERO_VALUE :
                super.setBackgroundDrawable(ContextCompat.getDrawable(getContext(), R.drawable.null_box));
                break ;
            case ButtonConstants.FLAG :
                super.setBackgroundDrawable(ContextCompat.getDrawable(getContext(), R.drawable.flag));
                break ;
            case ButtonConstants.BOMB :
                super.setBackgroundDrawable(ContextCompat.getDrawable(getContext(), R.drawable.bombed));
                break ;
        }

    }
}

File: app/src/main/java/com/shamanth/minesweeper/MyButton.kt
package com.shamanth.minesweeper

import android.content.Context
import androidx.core.content.ContextCompat
import androidx.appcompat.widget.AppCompatButton
import com.example.shamanth.minesweeper.R

/**
 ** Created by Shamanth on 27-06-2020
 */
class MyButton(var c: Context?) : AppCompatButton(c) {
    var value = 0
    var isFlagged = false
    var isVisible = false
    var validFlagCounter = 0
    var row = 0
    var col = 0
    fun applyChanges(state: Int) {
        when (state) {
            ButtonConstants.DEFAULT -> super.setBackground(ContextCompat.getDrawable(context, R.drawable.button))
            ButtonConstants.FALSEFLAG, ButtonConstants.ZERO_VALUE -> super.setBackground(ContextCompat.getDrawable(context, R.drawable.null_box))
            ButtonConstants.FLAG -> super.setBackground(ContextCompat.getDrawable(context, R.drawable.flag))
            ButtonConstants.BOMB -> super.setBackground(ContextCompat.getDrawable(context, R.drawable.bombed))
        }
    }

}